src/geventhttpclient/client.py

```diff
@@ -156,9 +157,12 @@ def request(self, method, request_uri, body=b"", headers={}):
         while 1:
             sock = self._connection_pool.get_socket()
             try:
-                sock.sendall(request)
+                sock.sendall(request.encode())
                 if body:
-                    sock.sendall(body)
+                    if type(body) == six.text_type:
+                        sock.sendall(body.encode('iso-8859-1'))
```

Comment:
the force to latin1 is a bit arbitrary but maybe people are used to it.

If it's a file we are probably not handling it well right now. We should use transfer encoding chunked or close the connection when EOF if no Content-Length header is set.